### ISSUE TITLE
Use aws_launch_template instead of launch configuration?

### ISSUE THREAD

**Issue by itsdalmo** [telia-oss/terraform-aws-asg#11]
The new `t3` instances launch with unlimited mode being enabled by default, and there is no way to disable it using launch configurations. It seems one needs to use [aws_launch_template](https://www.terraform.io/docs/providers/aws/r/launch_template.html) to have control over the setting. Launch templates are as far as I can gather the "new launch configuration".

**Comment by itsdalmo** [comment]
Tested this as part of [0.12-refactoring](https://github.com/telia-oss/terraform-aws-asg/tree/0.12-refactoring) (code is currently in my stash). Ran into a number of bugs with `aws_launch_template` variables that do not support a `null` value from Terraform, which caused actual crashes. This was with version 2.12.0 of the AWS provider.

Also, for loops within template directives `%{}` seem to be causing a diff on every run, which makes 0.12.0 unusable for now. Will attempt again in a few releases.

**Comment by evanstachowiak** [comment]
Any updates on this now that 0.12 is merged?

**Comment by itsdalmo** [comment]
The terraform resource seemed very buggy when I last attempted to use it so I have not given it another try yet unfortunately. Is there a particular feature you are looking for in launch templates?

**Comment by evanstachowiak** [comment]
Yes, I would like to use the unlimited CPU credit setting which is only available with launch templates. In general, they're also just nicer to use and copy around if necessary.

I've been using them for a while now in other modules and have not seen any issues.

**Comment by itsdalmo** [comment]
Great to hear @evanstachowiak; seems like launch templates is where AWS wants us to go, so I think it would be great to make the switch if it finally works well in Terraform now 👍
